airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -364,19 +364,21 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
 
             job = self._submit_job(hook, job_id="")
             context["ti"].xcom_push(key="job_id", value=job.job_id)
-            self.defer(
-                timeout=self.execution_timeout,
-                trigger=BigQueryValueCheckTrigger(
-                    conn_id=self.gcp_conn_id,
-                    job_id=job.job_id,
-                    project_id=hook.project_id,
-                    sql=self.sql,
-                    pass_value=self.pass_value,
-                    tolerance=self.tol,
-                    poll_interval=self.poll_interval,
-                ),
-                method_name="execute_complete",
-            )
+            if job.running():
+                self.defer(
+                    timeout=self.execution_timeout,
+                    trigger=BigQueryValueCheckTrigger(
+                        conn_id=self.gcp_conn_id,
+                        job_id=job.job_id,
+                        project_id=hook.project_id,
+                        sql=self.sql,
+                        pass_value=self.pass_value,
+                        tolerance=self.tol,
+                        poll_interval=self.poll_interval,
+                    ),
+                    method_name="execute_complete",
+                )
+            self.log.info("Current state of job %s is %s", job.job_id, job.state)
```

Comment:
This will only be logged if the job ended before the if block, since `defer` ends the execution immediately. Is this intended?

Reply:
yes it is intended, because otherwise if the job ended before getting deferred, the user wont know it's status

Reply:
Hmm, how will user know the status from Airflow otherwise? unless this is added, they need to see in BigQuery console